### DISCUSSION TITLE
vscode-twind-intellisense breaks TS type checking and does not provide any functionality

### DISCUSSION THREAD

**Discussion by jmayergit** [tw-in-js/twind#444] [Q&A]
Difficult to reproduce this anywhere but my machine however,

- TS version 4.9.4
- VS Code 1.74.3

When I enable the extension Type checking disappears. Also no functionality is provided by the extension, like template literal autocomplete. I've had issues with this extension in another project as well, any trouble shooting tips are appreciated. I've done things like uninstall a few (not all) extensions, re install vs code, restart my computer, etc. I know that this extension has plans to be overhauled but in the meantime..

https://user-images.githubusercontent.com/8594272/215936182-03cf847c-c409-4ab7-90d4-81cc55f81d8f.mov

My VS Code output for the extension

```
[11:59:17 AM] Extension Name: sastan.twind-intellisense.
[11:59:17 AM] Extension Version: 0.2.1.
[11:59:17 AM] Using local twind
[11:59:17 AM] Extension is enabled. To disable, change the `twind.enable` setting to `false`.
[11:59:17 AM] Configuring @twind/typescript-plugin using: {
  "tags": [
    "tw",
    "apply"
  ],
  "attributes": [
    "tw",
    "class",
    "className"
  ],
  "styles": [
    "style",
    "styled"
  ],
  "debug": false,
  "enable": true
}
```

**Comment by sastan** [comment]
Thanks for the heads-up and the time for testing.

 I really don't know what is happing there. The issue seems to be on the twind-plugin side.

I try to find time to investigate.

**Comment by jmayergit** [comment]
Thanks, I forgot to mention that I am using a twind.config.ts file. When I toggle the tailwind preset on and off in my configuration, TS type checking also toggles on and off. Usually a `> Developer: Reload Window` is required.

```
import { defineConfig } from '@twind/core'
import presetTailwind from '@twind/preset-tailwind'

export default defineConfig({
  // presets: [presetTailwind()],
})
```

This same behavior occurs separate projects I've worked on involving twind.

**In summary** the combination of the `presetTailwind()` + extension prevent TS type checking. Regardless no functionality is provided by the extension
